Make Lambda function instruction set architecture variable

jeohist opened the pull request:
## Description of the change

> [ec2-workerpool-scheduler](https://github.com/spacelift-io/ec2-workerpool-autoscaler) releases an amd64 and arm64 version.
> By adding a variable to this module, users are given the choice which one they'd like to use.
> amd64 was chosen as the default as this was previously the case.

## Type of change

- [X] New feature (non-breaking change that adds functionality);

## Checklists

### Development

- [X] All necessary variables have been defined, with defaults if applicable;
- [X] The code is formatted properly;

### Code review

- [x] The module version is bumped accordingly;
- [x] Spacelift tests are passing;
- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached;
- [x] This pull request is no longer marked as "draft";
- [x] Reviewers have been assigned;
- [x] Changes have been reviewed by at least one other engineer;

jmfontaine reviewed (approved):
Good idea. Thanks for the contribution!

eminalemdar reviewed (changes requested):
to be able to introduce these changes the module version should be bumped as well both in the config and the documentation section. can you please also update those?

jeohist commented:
> to be able to introduce these changes the module version should be bumped as well both in the config and the documentation section. can you please also update those?

I've rebased the MR and updated the module version referenced with a patch update.

eminalemdar commented on .spacelift/config.yml:
```diff
@@ -1,5 +1,5 @@
 version: 1
-module_version: 2.1.0
+module_version: 2.1.1
```
This change is introducing a new functionality and I think updating the minor version would be more appropriate. Thank you!

```suggestion
module_version: 2.2.0
```

eminalemdar commented on README.md:
```diff
@@ -19,7 +19,7 @@ terraform {
 
 module "my_workerpool" {
-  source = "github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2?ref=v2.1.0"
+  source = "github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2?ref=v2.1.1"
```
This change is introducing a new functionality and I think updating the minor version would be more appropriate. Thank you!

```suggestion
  source = "github.com/spacelift-io/terraform-aws-spacelift-workerpool-on-ec2?ref=v2.2.0"
```

eminalemdar reviewed (approved):
LGTM! Thank you for your contribution @jeohist
